.eleventy.js

```diff
@@ -13,6 +13,68 @@ const pluginMermaid = require("@kevingimbel/eleventy-plugin-mermaid");
 const syntaxHighlight = require("@11ty/eleventy-plugin-syntaxhighlight");
 const inspect = require("util").inspect;
 const striptags = require("striptags");
+const path = require("path");
+const matter = require('gray-matter');
+
+
+function fileExists(filePath) {
+  return fs.existsSync(filePath);
+}
+
+function loadConfig(filePath) {
+  try {
+    const resolvedPath = path.resolve(filePath);
+    return require(resolvedPath);
+  } catch (error) {
+    return false;
+  }
+}
+
+function getFilesByTypes(directoryPath, fileTypes) {
+  const files = fs.readdirSync(directoryPath);
+  const matchingFiles = [];
+  for (const file of files) {
+    const filePath = path.join(directoryPath, file);
+    const fileStat = fs.statSync(filePath);
+    if (fileStat.isFile() && fileTypes.includes(path.extname(file).toLowerCase())) {
+      matchingFiles.push(file);
+    }
+  }
+  return matchingFiles;
+}
+
+function getFrontMatterSync(filePath) {
```

Comment:
please annotate these special functions with what they're intended to be used for, for future generations :)

Reply:
thanks!